collector/aws/elasticsearch_test.go

```diff
@@ -0,0 +1,159 @@
+				EBSEnabled: awsClient.Bool(true),
+				VolumeSize: awsClient.Int64(10),
+				VolumeType: awsClient.String("gp2"),
+			},
+		},
+	},
+}
+
+type MockAWSElasticSearchClient struct {
+	responseDescribeClusters *elasticsearch.DescribeElasticsearchDomainsOutput
+	err                      error
+}
+
+func (es *MockAWSElasticSearchClient) DescribeElasticsearchDomains(*elasticsearch.DescribeElasticsearchDomainsInput) (*elasticsearch.DescribeElasticsearchDomainsOutput, error) {
+	return es.responseDescribeClusters, es.err
+}
+
+func (es *MockAWSElasticSearchClient) ListDomainNames(*elasticsearch.ListDomainNamesInput) (*elasticsearch.ListDomainNamesOutput, error) {
+	return &elasticsearch.ListDomainNamesOutput{
+		DomainNames: []*elasticsearch.DomainInfo{
+			{
+				DomainName: awsClient.String("testDomain"),
+			},
+		},
+	}, es.err
+}
+
+func (es *MockAWSElasticSearchClient) ListTags(*elasticsearch.ListTagsInput) (*elasticsearch.ListTagsOutput, error) {
+
+	return &elasticsearch.ListTagsOutput{}, es.err
+}
+
+func TestDescribeElasticSearchClusters(t *testing.T) {
+
+	collector := testutils.NewMockCollector()
+	metrics := []config.MetricConfig{}
+
+	t.Run("valid", func(t *testing.T) {
+
+		mockClient := MockAWSElasticSearchClient{
+			responseDescribeClusters: &defaultElasticSearchMock,
+		}
+
+		esManager := aws.NewElasticSearchManager(collector, &mockClient, nil, nil, nil, metrics, "us-east-1")
+
+		result, _ := esManager.DescribeClusters()
+
+		if len(result) != len(defaultElasticSearchMock.DomainStatusList) {
+			t.Fatalf("unexpected elasticsearch clusters count, got %d expected %d", len(result), len(defaultElasticSearchMock.DomainStatusList))
+		}
+	})
+
+	t.Run("error", func(t *testing.T) {
+
+		mockClient := MockAWSElasticSearchClient{
+			responseDescribeClusters: &defaultElasticSearchMock,
+			err:                      errors.New("error"),
+		}
+
+		esManager := aws.NewElasticSearchManager(collector, &mockClient, nil, nil, nil, metrics, "us-east-1")
+
+		_, err := esManager.DescribeClusters()
+
+		if err == nil {
+			t.Fatalf("unexpected describe clusters error, returned empty")
+		}
+	})
+
+}
+
+func TestDetectElasticSearch(t *testing.T) {
+
+	collector := testutils.NewMockCollector()
+	mockCloudwatchClient := MockAWSCloudwatchClient{
+		responseMetricStatistics: defaultResponseMetricStatistics,
+	}
+
+	mockSTSClient := MockAWSSTSClient{
+		responseGetCallerIdentity: &defaultSTSGetCallerIdentity,
+	}
+	stsManager := aws.NewSTSManager(&mockSTSClient)
+	cloutwatchManager := aws.NewCloudWatchManager(&mockCloudwatchClient)
+	pricingManager := aws.NewPricingManager(&defaultPricingMock, "us-east-1")
+
+	mockClient := MockAWSElasticSearchClient{
+		responseDescribeClusters: &defaultElasticSearchMock,
+	}
+
+	esManager := aws.NewElasticSearchManager(collector, &mockClient, cloutwatchManager, pricingManager, stsManager, defaultMetricConfig, "us-east-1")
+
+	response, _ := esManager.Detect()
+
+	if len(response) != 1 {
```

Comment:
Can we check the response that bigger than 1? 
To check if the cloudwatch returns a used elasticsearch?

Reply:
I have added more types of ElasticSearch cluster to check more than 1 type.